draft-ietf-quic-transport.md

```diff
@@ -690,14 +690,16 @@ The client MUST choose a random connection ID and use it in Initial packets
 ({{packet-initial}}) and 0-RTT packets ({{packet-protected}}).
 
 When the server receives a Initial packet and decides to proceed with the
-handshake, it chooses a new value for the connection ID and sends that in a
-Handshake packet ({{packet-handshake}}).  The server MAY choose to use the value
-that the client initially selects.
+handshake, it may chooses a new value for the connection ID and sends that in
+the server_connection_id transport parameter ({{transport-parameter-definitions}})
+of the Handshake packet ({{packet-handshake}}).
 
-Once the client receives the connection ID that the server has chosen, it MUST
+If the client receives a new connection ID that the server has chosen, it MUST
```

Comment:
Transport parameters are only really available once the handshake completes.  For one, they are in TLS EncryptedExtensions, which isn't necessarily right up front.  More importantly, though we could release them before the handshake completes, it makes it harder to reason about their authenticity.  Given that transport parameters are used for client address validation, that's problematic.

Reply:
Aren't they available at the server side when a CHLO is received?  And shouldn't they be available in the stateless retry?